File: CSharp/models/Stalker.cs
using models;

namespace models
{
    public class Stalker : Cordyceps
    {
        // atributo protegido
        protected bool hidden;

        //constructor
        public Stalker(string origin) : base(origin)
        {
            hidden = true;
        }

        // metodo con retorno
        public override string Behavior()
        {
            return "En esta fase del cordyceps sigue siendo muy parecida a la anterior, pero en esta se oculta y ataca por sorpresa.";
        }

        //metodo sin retorno protegido
        protected void Hide()
        {
            Console.WriteLine("El acechador se ha escondido. Cuidado al acercarse.");
        }

        // metodo sin retorno que accede al protegido
        public void ActivateStealth()
        {
            Hide();
        }

        //funcion con retorno
        public bool IsHidden()
        {
            return hidden;
        }

        // funcion sin retorno
        public void Reveal()
        {
            hidden = false;
        }
    }
}

File: GO/models/stalker.go
package models

import "fmt"

type Stalker struct {
    BaseCordyceps
    Hidden bool // Atributo publico
}

//constructor
func NewStalker(origin string) *Stalker {
    return &Stalker{
        BaseCordyceps: BaseCordyceps{origin: origin},
        Hidden:        true,
    }
}

// metodo con retorno
func (s *Stalker) Behavior() string {
    return "En esta fase del cordyceps sigue siendo muy parecida a la anterior, pero en esta se oculta y ataca por sorpresa."
}

//metodo sin retorno privado
func (s *Stalker) hide() {
    fmt.Println("El acechador se ha escondido. Cuidado al acercarse.")
}

// metodo sin retorno público que accede al privado
func (s *Stalker) ActivateStealth() {
    s.hide()
}

//funcion con retorno
func (s *Stalker) IsHidden() bool {
    return s.Hidden
}

//funcion sin retorno
func (s *Stalker) Reveal() {
    s.Hidden = false
}
